### PR TITLE
feature: first version of light mode for app

### PR DESCRIPTION
**Pull request by amaransen**
Nostr ID connection page:
image

Topics page:
image

Value split + Value entry:
image

Initial Statements (Warrant unexpanded):
image

Warrant: I wasn't able to make changes to the warrant button b/c I couldn't figure it out in a reasonable amt of time:
image

Create warrant: 
image

Rebuttal: Added blue highlighting to replace the grey
image

Highlighting on the counter argument + Summary:
image

AFAIK the only thing that wasn't addressed is the highlighting on the `create warrant` button and making light mode the default. I also need to change the SVG on the topic page to the new one.

### CODE REVIEW

**Comment by cdxker** [comment]
Look really good only thing is that I think we should standardize how the profile picture view looks
aside from that looks good to me

image

The more I look at it the more I feel like we should also change the color of the aff/neg border from the dark red/dark green to something lighter

File: src/components/Atoms/Combobox.tsx [hunk 1]
```diff
@@ -100,7 +100,7 @@ export const Combobox = (props: ComboboxProps) => {
           }}
         >
           {props.aboveOptionsElement}
-          <Menu class="flex w-full flex-col space-y-1 overflow-y-auto bg-gray-800 shadow-lg ring-1 ring-black ring-opacity-5 overflow-x-hidden">
+          <Menu class="flex w-full rounded flex-col space-y-1 overflow-y-auto bg-pink-50 dark:bg-gray-800 shadow-lg ring-1 ring-black ring-opacity-5 overflow-x-hidden">
```
Comment (skeptrunedev) [inline src/components/Atoms/Combobox.tsx]:
```suggestion
          <Menu class="flex w-full flex-col space-y-1 overflow-y-auto rounded bg-pink-50 shadow-lg ring-1 ring-black ring-opacity-5 overflow-x-hidden dark:bg-gray-800">
```

Linter wants a different class order

File: src/components/Atoms/InputRowsForm.tsx [hunk 2]
```diff
@@ -23,8 +23,7 @@ const InputRowsForm = (props: InputRowsFormProps) => {
   });
 
   return (
-    <div
-      class={`w-full rounded-lg border ${borderColor()} px-2 py-2 text-white`}
+    <div class={`w-full rounded border dark:${borderColor()} px-2 py-2 border-fuchsia-400 dark:text-white `}
```
Comment (skeptrunedev) [inline src/components/Atoms/InputRowsForm.tsx]:
```suggestion
    <div
      class={`w-full rounded border dark:${borderColor()} border-fuchsia-400 px-2 py-2 dark:text-white `}
    >
```

File: src/components/Atoms/InputRowsForm.tsx [hunk 3]
```diff
@@ -35,7 +34,7 @@ const InputRowsForm = (props: InputRowsFormProps) => {
                 {((!inputGroup.component && !inputGroup.type) ||
                   inputGroup.type == "input") && (
                   <input
-                    class="w-full rounded border border-white bg-slate-900 px-2 text-white"
+                    class="w-full rounded border border-fuchsia-300 dark:border-white dark:bg-slate-900 px-2 dark:text-white"
```
Comment (skeptrunedev) [inline src/components/Atoms/InputRowsForm.tsx]:
```suggestion
                    class="w-full rounded border border-fuchsia-300 px-2 dark:border-white dark:bg-slate-900 dark:text-white"
```

File: src/components/Atoms/InputRowsForm.tsx [hunk 4]
```diff
@@ -45,7 +44,7 @@ const InputRowsForm = (props: InputRowsFormProps) => {
                 )}
                 {!inputGroup.component && inputGroup.type == "textarea" && (
                   <textarea
-                    class="w-full rounded border border-white bg-slate-900 px-2 text-white"
+                    class="w-full rounded border border-fuchsia-300 dark:border-white dark:bg-slate-900 px-2 text-black dark:text-white"
```
Comment (skeptrunedev) [inline src/components/Atoms/InputRowsForm.tsx]:
```suggestion
                    class="w-full rounded border border-fuchsia-300 px-2 text-black dark:border-white dark:bg-slate-900 dark:text-white"
```

File: src/components/CounterArgument/CounterArgumentView.tsx [hunk 5]
```diff
@@ -22,10 +22,10 @@ export const CounterArgumentView = (props: CounterArgumentViewProps) => {
   return (
     <div
       classList={{
-        "flex flex-col space-y-2 rounded-md border-2 p-2 text-white": true,
+        "flex flex-col space-y-2 rounded-md border-2 p-2 text-black dark:text-white": true,
```
Comment (skeptrunedev) [inline src/components/CounterArgument/CounterArgumentView.tsx]:
```suggestion
       "flex flex-col space-y-2 rounded-md border-2 p-2 text-black dark:text-white":
          true,
```

File: src/components/Statements/AddStatementButton.tsx [hunk 6]
```diff
@@ -12,7 +12,7 @@ export const AddStatementButton = (props: {
         <div class="flex w-full justify-center">
           <div
             onClick={() => props.setShowStatementForm(true)}
-            class="flex w-fit cursor-pointer items-center space-x-2 rounded-lg border-2 border-white px-4 py-3 text-white"
+            class="flex w-fit cursor-pointer items-center space-x-2 rounded-lg border-2 border-blue-500 dark:border-white px-4 py-3 text-blue-600 dark:text-white"
```
Comment (skeptrunedev) [inline src/components/Statements/AddStatementButton.tsx]:
```suggestion
            class="flex w-fit cursor-pointer items-center space-x-2 rounded-lg border-2 border-blue-500 px-4 py-3 text-blue-600 dark:border-white dark:text-white"
```

File: src/components/Topics/TopicsDisplay.tsx [hunk 7]
```diff
@@ -52,12 +52,12 @@ const TopicsDisplay = () => {
...
-    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-slate-600 py-2 md:max-w-[75%]">
-      <div class="flex w-full items-center justify-center space-x-3 border-b border-slate-700 py-2">
-        <div class="w-fit text-2xl font-bold text-white">Your Topics</div>
+    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-violet-400 dark:border-slate-600 py-2 md:max-w-[75%]">
+      <div class="flex w-full items-center justify-center space-x-3 border-b border-violet-400 dark:border-slate-700 py-2">
+        <div class="w-fit text-2xl font-bold text-blue-700 dark:text-white">Your Topics</div>
         {!showCreateTopicForm() && (
           <button
-            class="rounded-full border border-green-500 bg-transparent p-2 text-green-500"
+            class="rounded-full border border-teal-300 dark:border-green-700 bg-transparent p-2 text-green-500"
```
Comment (skeptrunedev) [inline src/components/Topics/TopicsDisplay.tsx]:
```suggestion
            class="rounded-full border border-teal-300 bg-transparent p-2 text-green-500 dark:border-green-700"
```

File: src/components/Topics/TopicsDisplay.tsx [hunk 8]
```diff
@@ -52,12 +52,12 @@ const TopicsDisplay = () => {
   };
 
   return (
-    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-slate-600 py-2 md:max-w-[75%]">
-      <div class="flex w-full items-center justify-center space-x-3 border-b border-slate-700 py-2">
-        <div class="w-fit text-2xl font-bold text-white">Your Topics</div>
+    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-violet-400 dark:border-slate-600 py-2 md:max-w-[75%]">
+      <div class="flex w-full items-center justify-center space-x-3 border-b border-violet-400 dark:border-slate-700 py-2">
+        <div class="w-fit text-2xl font-bold text-blue-700 dark:text-white">Your Topics</div>
```
Comment (skeptrunedev) [inline src/components/Topics/TopicsDisplay.tsx]:
```suggestion
        <div class="w-fit text-2xl font-bold text-blue-700 dark:text-white">
          Your Topics
        </div>
```

File: src/components/Topics/TopicsDisplay.tsx [hunk 9]
```diff
@@ -52,12 +52,12 @@ const TopicsDisplay = () => {
   };
 
   return (
-    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-slate-600 py-2 md:max-w-[75%]">
-      <div class="flex w-full items-center justify-center space-x-3 border-b border-slate-700 py-2">
-        <div class="w-fit text-2xl font-bold text-white">Your Topics</div>
+    <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-violet-400 dark:border-slate-600 py-2 md:max-w-[75%]">
+      <div class="flex w-full items-center justify-center space-x-3 border-b border-violet-400 dark:border-slate-700 py-2">
```
Comment (skeptrunedev) [inline src/components/Topics/TopicsDisplay.tsx]:
```suggestion
        <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-violet-400 py-2 dark:border-slate-600 md:max-w-[75%]">
      <div class="flex w-full items-center justify-center space-x-3 border-b border-violet-400 py-2 dark:border-slate-700">
```

File: src/components/Topics/TopicsList.tsx [hunk 10]
```diff
@@ -111,13 +111,13 @@ const TopicsList = () => {
...
           return (
-            <div class="w-full rounded-lg bg-gray-800">
+            <div class="w-full rounded-lg bg-emerald-200 dark:bg-gray-800">
               <A
                 // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
                 href={`/topics/${topic.event.id}`}
                 aria-label="topic detail page"
               >
-                <div class="p-4 text-lg font-bold text-white">
+                <div class="p-4 text-lg font-bold text-blue-600 dark:text-white border rounded-lg border-violet-400">
```
Comment (skeptrunedev) [inline src/components/Topics/TopicsList.tsx]:
```suggestion
                <div class="rounded-lg border border-violet-400 p-4 text-lg font-bold text-blue-600 dark:text-white">
```

File: src/components/Topics/ValueSplitButton.tsx [hunk 11]
```diff
@@ -73,7 +73,7 @@ const ValueSplitButton = (props: ValueSplitButtonProps) => {
               </div>
               <Show when={isOpen()}>
                 <PopoverPanel class="absolute z-10 w-full" unmount={false}>
-                  <Menu class="left-0 mt-1 flex w-full origin-top-left flex-col space-y-1 rounded-lg bg-gray-800 p-1 py-2 shadow-lg ring-1 ring-black ring-opacity-5">
+                  <Menu class="left-0 mt-1 flex w-full origin-top-left flex-col space-y-1 rounded-lg bg-fuchsia-200 dark:bg-gray-800 p-1 py-2 shadow-lg ring-1 ring-black ring-opacity-5">
```
Comment (skeptrunedev) [inline src/components/Topics/ValueSplitButton.tsx]:
```suggestion
                  <Menu class="left-0 mt-1 flex w-full origin-top-left flex-col space-y-1 rounded-lg bg-fuchsia-200 p-1 py-2 shadow-lg ring-1 ring-black ring-opacity-5 dark:bg-gray-800">
```

File: src/components/layouts/ApplicationLayout.tsx [hunk 12]
```diff
@@ -10,12 +10,12 @@ const ApplicationLayout = (props: ApplicationLayoutProps) => {
   const globalContext = useContext(GlobalContext);
 
   return (
-    <div class="min-h-screen w-screen bg-black overflow-x-hidden">
+    <div class="min-h-screen w-screen bg-pink-50 dark:bg-black overflow-x-hidden">
```
Comment (skeptrunedev) [inline src/components/layouts/ApplicationLayout.tsx]:
```suggestion
    <div class="min-h-screen w-screen bg-pink-50 overflow-x-hidden dark:bg-black">
```

File: src/routes/topics/[id].tsx [hunk 13]
```diff
@@ -272,7 +272,7 @@ const TopicDetail = () => {
     <ApplicationLayout>
       <div class="flex flex-col space-y-4 px-4 py-2 md:px-8 md:py-4">
         <div class="mt-4 flex w-full justify-center">
-          <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-slate-600 p-2 md:px-8 md:py-4">
+          <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-emerald-300 dark:border-slate-600 p-2 md:px-8 md:py-4">
```
Comment (skeptrunedev) [inline src/routes/topics/[id].tsx]:
```suggestion
          <div class="flex w-full flex-col items-center justify-center space-y-6 rounded-lg border border-emerald-300 p-2 dark:border-slate-600 md:px-8 md:py-4">
```

File: src/root.tsx [hunk 14]
```diff
@@ -14,14 +14,17 @@ import "./root.css";
 import RelayStoreContext from "./contexts/GlobalContext";
 
 export default function Root() {
+  let setDark = ""
+  if(localStorage.theme === 'dark' || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches))
+    setDark = "dark";
```
Comment (skeptrunedev) [inline src/root.tsx]:
```suggestion
  let setDark = "";
  if (
    localStorage.theme === "dark" ||
    (!("theme" in localStorage) &&
      window.matchMedia("(prefers-color-scheme: dark)").matches)
  )
    setDark = "dark";
```

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
Please add the prettier and eslint extensions to your VSCode. 

I'd like to fix the linting errors and check if there are any other ones before merging this. 

I'd also like to make sure we ship the controller button in the navbar before merging this. 

In summary: 

- [ ] prettier and eslint checks
- [ ] controller button in the navbar

**Pull request merged**